greenmail-core/src/main/java/com/icegreen/greenmail/imap/commands/FetchCommand.java

```diff
@@ -37,6 +48,7 @@
     public static final String ARGS = "<message-set> <fetch-profile>";
     private static final Flags FLAGS_SEEN = new Flags(Flags.Flag.SEEN);
     private static final Pattern NUMBER_MATCHER = Pattern.compile("^\\d+$");
+    private static final SimpleDateFormat INTERNALDATE_FORMAT = new SimpleDateFormat("dd-MMM-yyyy HH:mm:ss Z");

```

Comment:
SimpleDateFormat is not [threadsafe](https://docs.oracle.com/javase/7/docs/api/java/text/SimpleDateFormat.html#synchronization) - therefore making it static is a bad idea